dbms/src/Functions/FunctionsRegexpCommon.h

```diff
@@ -102,19 +102,34 @@ inline int getDefaultFlags()
     return flags;
 }
 
+template <bool need_subpattern = false>
 inline String addMatchTypeForPattern(const String & pattern, const String & match_type, TiDB::TiDBCollatorPtr collator)
 {
+    if (pattern.empty())
+        throw Exception("Length of the pattern argument must be greater than 0.");
+
     String mode = re2Util::getRE2ModeModifiers(match_type, collator);
+    String final_pattern;
+    if constexpr (need_subpattern)
+        final_pattern = fmt::format("({})", pattern);
```

Comment:
What happens if `pattern` itself is already inside parentheses? Like `pattern = (xxx)`

Reply:
> What happens if `pattern` itself is already inside parentheses? Like `pattern = (xxx)`

When we use the `FindAndConsume` function in re2, we need this pattern, or there is no way to get the matched string.